internal/pkg/service/common/etcdop/op/atomic.go

```diff
@@ -250,44 +250,49 @@ func (v *AtomicOp[R]) DoWithoutRetry(ctx context.Context, opts ...Option) *TxnRe
 	// Create IF part of the transaction
 	var cmps []etcd.Cmp
 	for _, op := range removeOpsOverlaps(tracker.Operations()) {
-		switch op.Type {
-		case DeleteOp:
+		mustExist := (op.Type == GetOp || op.Type == PutOp) && op.Count > 0
+		mustNotExist := op.Type == DeleteOp || op.Count == 0
+		switch {
+		case mustExist:
+			// IF: 0 < modification version <= Read Phase revision
+			// Key/range exists and has not been modified since the Read Phase.
+			//
```

Comment:
Original implementation didin't covered all cases. Original tests have been keept, but new tests cover all possible combinations of operations.

Changes:


**1. If the key does not exist, it can be written in different ways:**
  - `MOD == 0`, `VERSIO == 0`, `CREATE == 0`.
  - I replaced `VERSION == 0` by `MOD == 0`, because `MOD` is used also in other conditions.

**2. `put` method in Read Phase must generate also confition "key must exists", I have no idea why it was missing there.**
https://github.com/keboola/keboola-as-code/blob/317d484d8d1cfab4c96f91b636e4727c8a3c8c5c/internal/pkg/service/common/etcdop/op/atomic.go#L167-L177

Reply:
You can easily review it as a new code, the original tests remained.